presto-orc/src/main/java/com/facebook/presto/orc/StripeReader.java

```diff
@@ -266,8 +275,15 @@ public Stripe readStripe(StripeInformation stripe, AggregatedMemoryContext syste
         // transform streams to OrcInputStream
```

Comment:
There is a `readFully` above, which I don't think it should appear over there? We will need to figure out what `diskRanges` to be read (e.g., col1 and col3 are new; while col2 and col4 are already cached) and put them into the cache accordingly.

Reply:
I move range transformation and read logic into `StripeMetadataSource`, thought after this I'm not really sure if this should be called `StripeMetadataSource` any more :(